File: developer/Samples/NET 4.6/WF_WCF_Samples/WCF/Basic/Binding/ByteStreamEncoder/CS/ByteStreamHttpBinding/AddressValidatorAttribute.cs
//----------------------------------------------------------------
// Copyright (c) Microsoft Corporation.  All rights reserved.
//----------------------------------------------------------------

using System;
using System.Configuration;

namespace Microsoft.Samples.ByteStreamEncoder
{

    [AttributeUsage(AttributeTargets.Property)]
    internal sealed class AddressValidatorAttribute : ConfigurationValidatorAttribute
    {
        public override ConfigurationValidatorBase ValidatorInstance
        {
            get { return new AddressValidator(); }
        }
    }
}

File: developer/Samples/NET 4.6/WF_WCF_Samples/WCF/Basic/Binding/ByteStreamEncoder/VB/ByteStreamHttpBinding/AddressValidatorAttribute.vb
'----------------------------------------------------------------
' Copyright (c) Microsoft Corporation.  All rights reserved.
'----------------------------------------------------------------

Imports Microsoft.VisualBasic
Imports System
Imports System.Configuration

Namespace Microsoft.Samples.ByteStreamEncoder

    <AttributeUsage(AttributeTargets.Property)> _
    Friend NotInheritable Class AddressValidatorAttribute
        Inherits ConfigurationValidatorAttribute

        Public Overrides ReadOnly Property ValidatorInstance() As ConfigurationValidatorBase
            Get
                Return New AddressValidator()
            End Get
        End Property
    End Class
End Namespace
